Including common.h with NEON_2_SSE.h

slevental opened the pull request:
Including common.h to make sure that USE_NEON is defined in case of NEON_2_SSE.h is used; otherwise USE_NEON will not be propagated to this file and `portable_tensor_utils.h` will be used

slevental committed:
Including common.h with NEON_2_SSE.h

Including common.h to make sure that USE_NEON is defined in case of NEON_2_SSE.h is used; otherwise USE_NEON will not be propagated to this file and `portable_tensor_utils.h` will be used

slevental commented:
I signed it!

slevental committed:
Removing spaces

slevental committed:
Using neon is USE_NEON is defined

slevental committed:
Using common.h to set USE_NEON if applicable

drpngx commented:
Jenkins, test this please.

tensorflow-jenkins commented:
Can one of the admins verify this patch?

drpngx commented:
This does not build in all configurations:
```
./tensorflow/contrib/lite/kernels/internal/common.h:42:10: fatal error: 'NEON_2_SSE.h' file not found
#include "NEON_2_SSE.h"
```
while building `tensor_utils`.

slevental committed:
Adding required dependencies

Since tests are more likely compiled on sse4-enabled machine - if they are run with common.h - it will enable sse4, there are two ways - just enable it and let tests using sse4, or add extra flag check which would require USE_SSE4 flag to be defined. I've added all required dependencies for tests to pass, so it could work.

drpngx commented:
Jenkins, test this please.

slevental committed:
Rearranging dependencies according to linter

drpngx commented:
Jenkins, test this please.

slevental committed:
Fixing sanity check

drpngx commented:
Jenkins, test this please.

drpngx commented:
Still no go on MacOS.

slevental committed:
trying to rerun, could be cached

slevental committed:
Triggering build

slevental commented:
I've tried to build those tests on my MacOS and everything was fine, I've used similar parameters to the build that failed, so I think that it was related to caching or something like that, so I wanted to rerun builds to make sure that it's not the case

slevental committed:
triggering build - looks like it stuck

no status for more than 24h

drpngx commented:
Jenkins, test this please.

drpngx commented:
MacOS link error. Can you check?

```
ERROR: /Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/contrib/lite/kernels/BUILD:139:1: Couldn't build file tensorflow/contrib/lite/kernels/activations_test: Linking of rule '//tensorflow/contrib/lite/kernels:activations_test' failed (Exit 1): cc_wrapper.sh failed: error executing command 
  (cd /Volumes/BuildData/tmpfs/tmp/bazel/execroot/org_tensorflow && \
  exec env - \
    APPLE_SDK_PLATFORM=MacOSX \
    APPLE_SDK_VERSION_OVERRIDE=10.12 \
    PATH=/Volumes/BuildData/tmpfs/src/tf_build_env/bin:/Users/kbuilder/bin:/Users/kbuilder/.rvm/gems/ruby-2.4.1/bin:/Users/kbuilder/.rvm/gems/ruby-2.4.1@global/bin:/Users/kbuilder/.rvm/rubies/ruby-2.4.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/CMake.app/Contents/bin:/Applications/Unity/Unity.app/MacOS:/Users/kbuilder/Library/Android/sdk/platform-tools:/Users/kbuilder/Library/Android/sdk/tools/bin:/usr/local/sbin:/Users/kbuilder/.rvm/bin \
    PYTHON_BIN_PATH=/Volumes/BuildData/tmpfs/src/tf_build_env/bin/python2 \
    PYTHON_LIB_PATH=/Volumes/BuildData/tmpfs/src/tf_build_env/lib/python2.7/site-packages \
    TF_NEED_CUDA=0 \
    TF_NEED_OPENCL_SYCL=0 \
    TMPDIR=/tmpfs/tmp \
    XCODE_VERSION_OVERRIDE=8.1.0 \
  external/local_config_cc/cc_wrapper.sh -fobjc-link-runtime -o bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/activations_test '-Wl,-rpath,$ORIGIN/../../../../_solib_darwin_x86_64/_U_S_Stensorflow_Scontrib_Slite_Skernels_Cactivations_Utest___Utensorflow' -Lbazel-out/darwin_x86_64-opt/bin/_solib_darwin_x86_64/_U_S_Stensorflow_Scontrib_Slite_Skernels_Cactivations_Utest___Utensorflow bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/_objs/activations_test/tensorflow/contrib/lite/kernels/activations_test.o -ltensorflow_framework bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/libtest_util.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/libbuiltin_ops.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/internal/libquantization_util.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/internal/libtensor_utils.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/internal/libportable_tensor_utils.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/libstring_util.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/libframework.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/kernels/libgemm_support.a bazel-out/darwin_x86_64-opt/bin/tensorflow/contrib/lite/libcontext.a bazel-out/darwin_x86_64-opt/bin/external/com_google_absl/absl/base/libbase.a bazel-out/darwin_x86_64-opt/bin/external/com_google_absl/absl/base/libspinlock_wait.a bazel-out/darwin_x86_64-opt/bin/external/com_google_absl/absl/base/libdynamic_annotations.a bazel-out/darwin_x86_64-opt/bin/external/gif_archive/libgif.a bazel-out/darwin_x86_64-opt/bin/external/jpeg/libjpeg.a bazel-out/darwin_x86_64-opt/bin/external/jpeg/libsimd_none.a bazel-out/darwin_x86_64-opt/bin/external/com_googlesource_code_re2/libre2.a bazel-out/darwin_x86_64-opt/bin/external/farmhash_archive/libfarmhash.a bazel-out/darwin_x86_64-opt/bin/external/fft2d/libfft2d.a bazel-out/darwin_x86_64-opt/bin/external/highwayhash/libsip_hash.a bazel-out/darwin_x86_64-opt/bin/external/highwayhash/libarch_specific.a bazel-out/darwin_x86_64-opt/bin/external/png_archive/libpng.a bazel-out/darwin_x86_64-opt/bin/external/zlib_archive/libzlib.a bazel-out/darwin_x86_64-opt/bin/external/com_google_googletest/libgtest.a -lpthread -lm -Wl,-rpath,@loader_path/,-rpath,@loader_path/..,-rpath,@loader_path/../..,-rpath,@loader_path/../../.. -lpthread -ldl -ldl -lpthread -pthread -lm -lm -pthread -headerpad_max_install_names -lc++ -no-canonical-prefixes).
clang: warning: argument unused during compilation: '-pthread'
clang: warning: argument unused during compilation: '-pthread'
Undefined symbols for architecture x86_64:
  "tflite::tensor_utils::NeonClipVector(float const*, int, float, float*)", referenced from:
      tflite::tensor_utils::ClipVector(float const*, int, float, float*) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonSub1Vector(float const*, int, float*)", referenced from:
      tflite::tensor_utils::Sub1Vector(float const*, int, float*) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonVectorShiftLeft(float*, int, float)", referenced from:
      tflite::tensor_utils::VectorShiftLeft(float*, int, float) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonReductionSumVector(float const*, float*, int, int)", referenced from:
      tflite::tensor_utils::ReductionSumVector(float const*, float*, int, int) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonVectorVectorDotProduct(float const*, float const*, int)", referenced from:
      tflite::tensor_utils::VectorVectorDotProduct(float const*, float const*, int) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonVectorVectorCwiseProduct(float const*, float const*, int, float*)", referenced from:
      tflite::tensor_utils::VectorVectorCwiseProduct(float const*, float const*, int, float*) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonBatchVectorBatchVectorDotProduct(float const*, float const*, int, int, float*, int)", referenced from:
      tflite::tensor_utils::BatchVectorBatchVectorDotProduct(float const*, float const*, int, int, float*, int) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonVectorVectorCwiseProductAccumulate(float const*, float const*, int, float*)", referenced from:
      tflite::tensor_utils::VectorVectorCwiseProductAccumulate(float const*, float const*, int, float*) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonMatrixBatchVectorMultiplyAccumulate(float const*, int, int, float const*, int, float*, int)", referenced from:
      tflite::tensor_utils::MatrixBatchVectorMultiplyAccumulate(float const*, int, int, float const*, int, float*, int) in libtensor_utils.a(tensor_utils.o)
  "tflite::tensor_utils::NeonVectorBatchVectorCwiseProductAccumulate(float const*, int, float const*, int, float*)", referenced from:
      tflite::tensor_utils::VectorBatchVectorCwiseProductAccumulate(float const*, int, float const*, int, float*) in libtensor_utils.a(tensor_utils.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

rmlarsen commented:
@slevental any progress?

slevental committed:
including x86_64 since NEON_2_SSE4 convertion will allow neon to run on x86 with SSE3,4

slevental committed:
Fixing name

rmlarsen commented:
@drpngx wanna take one more look?

slevental commented:
@rmlarsen is the reason for failure the same, or it's a label that's missing causing it?

rmlarsen commented:
@slevental it looks like an internal infrastructure failure unrelated to your code. I'll run the tests again.

slevental commented:
Is there any way to see compiler parameters and what architecture it's using for this build, is it x86_64?

rmlarsen commented:
You should be able to click through the links for the failed build and see the logs. I cannot see them at the moment due to a network outage. I'll try again later.

rmlarsen commented:
The error seems unrelated to your change:

```
======================================================================
ERROR: test_long_eval (__main__.ARModelTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1350, in _do_call
    return fn(*args)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1329, in _run_fn
    status, run_metadata)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/framework/errors_impl.py", line 516, in __exit__
    c_api.TF_GetCode(self.status.status))
tensorflow.python.framework.errors_impl.NotFoundError: PruneForTargets: Some target nodes not found: init 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmpfs/tmp/bazel/execroot/org_tensorflow/bazel-out/k8-py3-opt/bin/bazel_pip/tensorflow/contrib/timeseries/python/timeseries/ar_model_test.runfiles/org_tensorflow/bazel_pip/tensorflow/contrib/timeseries/python/timeseries/ar_model_test.py", line 279, in test_long_eval
    variables.global_variables_initializer().run()
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 2281, in run
    _run_using_default_session(self, feed_dict, self.graph, session)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/framework/ops.py", line 4932, in _run_using_default_session
    session.run(operation, feed_dict)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 895, in run
    run_metadata_ptr)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1128, in _run
    feed_dict_tensor, options, run_metadata)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1344, in _do_run
    options, run_metadata)
  File "/tmpfs/src/github/tensorflow/tf_build_env/lib/python3.5/site-packages/tensorflow/python/client/session.py", line 1363, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.NotFoundError: PruneForTargets: Some target nodes not found: init 

----------------------------------------------------------------------
Ran 12 tests in 40.883s

FAILED (errors=1)
```

slevental committed:
using darwing for MacOS

drpngx commented:
It failed the package inclusion rule:
```
ERROR: /Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/contrib/lite/kernels/internal/BUILD:264:1: Couldn't build file tensorflow/contrib/lite/kernels/internal/_objs/neon_tensor_utils/tensorflow/contrib/lite/kernels/internal/optimized/neon_tensor_utils.o: undeclared inclusion(s) in rule '//tensorflow/contrib/lite/kernels/internal:neon_tensor_utils':
this rule is missing dependency declarations for the following files included by 'tensorflow/contrib/lite/kernels/internal/optimized/neon_tensor_utils.cc':
  '/Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/contrib/lite/kernels/internal/common.h'
  '/Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/contrib/lite/kernels/internal/types.h'
  '/Volumes/BuildData/tmpfs/src/github/tensorflow/tensorflow/contrib/lite/kernels/internal/compatibility.h'.
```

rmlarsen commented:
Ah, thanks @drpngx !

slevental committed:
Fixing header inclusion rules

drpngx commented:
Ugh, this is not really the way we do this, but I guess it's consistent with usage. FTR, normally we don't include the same file twice in a rule. Typically, only a single rule includes a file. For instance, `:dtypes` appears to be a rule that's "logical" for `compatibility.h`. So, you would depend on that library.

drpngx committed:
Fix syntax

drpngx committed:
Depend on includes

We depend on both types and compatibility. That's in the "types" rule.

slevental commented:
@drpngx thanks for your help!

drpngx commented:
Yay!
